app/kafka/manager/BrokerViewCacheActor.scala

```diff
@@ -143,6 +179,7 @@ class BrokerViewCacheActor(config: BrokerViewCacheActorConfig) extends LongRunni
   }
 
   private[this] def updateView(): Unit = {
+    var nBrokers: Int = 0
```

Comment:
This is not needed.  You can fix the missing brokers from the list by modifying lines 157 - 161.  You can keep an empty broker view:
```
val EMPTY_BVVIEW = BVView(Map.empty, config.clusterConfig, Option(BrokerMetrics.DEFAULT)).
```

And then change the following on line 157 :
```
        for {
          bv <- brokerTopicPartitions.get(id)
        } {
          brokerTopicPartitions.put(id, bv.copy(metrics = Option(metrics)))
        }
```

To something like:
```
          val updatedBVView = brokerTopicPartitions.getOrElse(id, EMPTY_BVVIEW).copy(metrics = Option(metrics))
          brokerTopicPartitions.put(id, updatedBVView)
 ```